lib/src/element_type.dart

```diff
@@ -50,10 +54,8 @@ abstract class ElementType extends Privacy with CommentReferable, Nameable {
       var isGenericTypeAlias = f.aliasElement != null && f is! InterfaceType;
       if (f is FunctionType) {
```

Comment:
Ah, this is probably what you mean about being backward compatible.
Do we need this though? Why not check for any `FunctionType`, with or without element and handle as `FunctionTypeElementType` or `AliasedFunctionTypeElementType`, and never as `CallableElementType`?

Reply:
Oh, perhaps you meant, why is `CallableElementType` needed ever?  Because that case uses the `DefinedElementType` base class instead of `UndefinedElementType`, and that has implications for some of the other getters in `ElementType`.  It wouldn't work to treat the two cases the same.